qubes/tests/__init__.py

```diff
@@ -383,6 +383,39 @@ def __exit__(self, exc_type, exc_value, tb):
         pkg_resources.iter_entry_points = self._orig_iter_entry_points
         self._orig_iter_entry_points = None
 
+def _clear_ex_info(exc_infos):
+    for exc_info in exc_infos:
+        if exc_info is None:
+            continue
+        ex = exc_info[1]
+        while ex is not None:
+            if isinstance(ex, qubes.exc.QubesVMError):
+                ex.vm = None
+            traceback.clear_frames(ex.__traceback__)
+            ex = ex.__context__
+
+class QubesTestResult(unittest.TestResult):
```

Comment:
I appreciate your attempt at cleaning this up, but a) it does not belong to this PR, and b) this most likely will fail with alternative runners as they may provide own result class (we do use `nose2` and sometimes `pytest` example).